addons/hr_holidays/models/resource.py

```diff
@@ -30,6 +33,123 @@ def _check_compare_dates(self):
                 if existing_leaves:
                     raise ValidationError(_('Two public holidays cannot overlap each other for the same working hours.'))
 
+    def _get_domain(self, time_domain_dict):
+        domain = []
+        for date in time_domain_dict:
+            domain = expression.OR([domain, [
+                    ('date_to', '>', date['date_from']),
+                    ('date_from', '<', date['date_to'])]
+            ])
+        return expression.AND([domain, [('state', '!=', 'refuse'), ('active', '=', True)]])
+
+    def _get_time_domain_dict(self):
+        return [{
+            'date_from' : record.date_from,
+            'date_to' : record.date_to
+        } for record in self if not record.resource_id]
+
+    def _split_leave_on_gto(self, leave, gto): #gto = global time off
+        leave_start = date_utils.start_of(leave.date_from, 'day')
+        leave_end = date_utils.end_of(leave.date_to - timedelta(seconds=1), 'day')
```

Comment:
I believe it's because I noticed some time off may be set with a date_to set as the next day at 0:00 am. By removing one second, I make sure to avoid taking the end of the wrong day